File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Prompt for the height until a valid height is given
    int in = -1;
    while (in > 23 || in < 0)
    {
        in = get_int("Height: ");
    }

    for (int i = 0; i < in; i++)
    {
        // Print spaces
        for (int j = i; j < in - 1; j++)
        {
            printf(" ");
        }
        // Print the first half of the pyramid
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("  ");
        // Print the second half of the pyramid
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: mario.py
from cs50 import get_int

# Gets a height from the user and check that it's valid
height = get_int("Height: ")
while height < 0 or height > 23:
    height = get_int("Height: ")

# Prints the pyramid
for i in range(height):
    for j in range(height - i - 1):
        print(end=" ")
    for j in range(i + 1):
        print(end="#")
    print(end="  ")
    for j in range(i + 1):
        print(end="#")
    print()
